tensorflow/python/data/util/sparse_test.py

```diff
@@ -18,58 +18,405 @@
 from __future__ import division
 from __future__ import print_function
 
+import functools
+
+from absl.testing import parameterized
+
+from tensorflow.python.data.kernel_tests import test_base
 from tensorflow.python.data.util import nest
 from tensorflow.python.data.util import sparse
+from tensorflow.python.framework import combinations
 from tensorflow.python.framework import constant_op
 from tensorflow.python.framework import dtypes
 from tensorflow.python.framework import ops
 from tensorflow.python.framework import sparse_tensor
 from tensorflow.python.framework import tensor_shape
-from tensorflow.python.framework import test_util
 from tensorflow.python.platform import test
 
 
-class SparseTest(test.TestCase):
-
-  def testAnySparse(self):
-    test_cases = (
-        {
-            "classes": (),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor),
-            "expected": False
-        },
-        {
-            "classes": (((ops.Tensor))),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor, ops.Tensor),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor, sparse_tensor.SparseTensor),
-            "expected": True
-        },
-        {
-            "classes": (sparse_tensor.SparseTensor, sparse_tensor.SparseTensor),
-            "expected":
-                True
-        },
-        {
-            "classes": (sparse_tensor.SparseTensor, ops.Tensor),
-            "expected": True
-        },
-        {
-            "classes": (((sparse_tensor.SparseTensor))),
-            "expected": True
-        },
+# NOTE(vikoth18): Arguments of parameterized tests are lifted into lambdas to make
```

Comment:
What happens when you do not use lambdas?

Reply:
I ran the tests without lambdas in `random_seed_test` (a similar situation to what we have in sparse_test).

If we do not use lambdas, then for the test combination of `(mode='graph', tf_api_version=2)`, the tests throw :
```console
File "/root/.cache/bazel/_bazel_root/f04f3979d0f8981d3682673d5f0c4874/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/data/util/random_seed_test.runfiles/org_tensorflow/tensorflow/python/framework/ops.py", line 1513, in convert_to_tensor
    raise RuntimeError("Attempting to capture an EagerTensor without "
RuntimeError: Attempting to capture an EagerTensor without building a function.
```

Reference: https://github.com/tensorflow/tensorflow/blob/f6957507f30597fd458028de1470010312b979f8/tensorflow/python/framework/ops.py#L1495-L1515

Due to this reason, The tests in `structure_test.py` were already using lambdas. Thus instead of splitting existing tests based on different execution modes, using lambdas can remove that redundancy as it works with all test_base combinations.

Reply:
thanks for the detailed response